File: src/main/java/no/nav/familie/ks/oppslag/dokarkiv/api/Dokument.java
package no.nav.familie.ks.oppslag.dokarkiv.api;

import javax.validation.constraints.NotEmpty;

public class Dokument {
    @NotEmpty
    private byte[] dokument;
    @NotEmpty
    private FilType filType;

    private String filnavn;

    @NotEmpty
    private DokumentType dokumentType;

    public Dokument(@NotEmpty byte[] dokument, @NotEmpty FilType filType, String filnavn, @NotEmpty DokumentType dokumentType) {
        this.dokument = dokument;
        this.filType = filType;
        this.filnavn = filnavn;
        this.dokumentType = dokumentType;
    }

    public byte[] getDokument() {
        return dokument;
    }

    public FilType getFilType() {
        return filType;
    }

    public String getFilnavn() {
        return filnavn;
    }

    public DokumentType getDokumentType() {
        return dokumentType;
    }
}

File: src/main/java/no/nav/familie/integrasjoner/dokarkiv/api/Dokument.kt
package no.nav.familie.integrasjoner.dokarkiv.api

import javax.validation.constraints.NotEmpty

class Dokument(@field:NotEmpty val dokument: ByteArray,
               @field:NotEmpty val filType: FilType,
               val filnavn: String?,
//               val dokumenttittel: String?,
               @field:NotEmpty val dokumentType: String)
